wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -485,19 +480,17 @@ class KotlinGenerator private constructor(
       }
 
       declarationBody.addStatement("%L", fieldDeclaration)
-      decodeBlock.addStatement(decodeBodyTemplate, indentation.repeat(2), field.tag(), fieldName,
-          adapterName)
-      returnBody.add("%L%L = %L%L,\n", indentation, fieldName, fieldName, throwExceptionBlock)
+      decodeBlock.addStatement(decodeBodyTemplate, field.tag(), fieldName, adapterName)
+      returnBody.addStatement("%L = %L%L,", fieldName, fieldName, throwExceptionBlock)
     }
 
     val tagHandlerClass = ClassName("com.squareup.wire", "TagHandler")
 
-    decodeBlock.addStatement("%Lelse -> %T.%L", indentation.repeat(2), tagHandlerClass,
-        "UNKNOWN_TAG")
-    decodeBlock.addStatement("%L}", indentation)
-    decodeBlock.addStatement("}")
+    decodeBlock.addStatement("else -> %T.%L", tagHandlerClass, "UNKNOWN_TAG")
+    decodeBlock.add("%<}\n".repeat(2)) // close the block
```

Comment:
Not sure I like the use of `repeat(2)` here. It's easy to miss.

Reply:
Fair point, can change